Infrastructure/TuitionPartnerService.cs

```diff
@@ -39,15 +115,22 @@ public IEnumerable<TuitionPartnerResult> OrderTuitionPartners(IEnumerable<Tuitio
 
             case TuitionPartnerOrderBy.MinPrice:
                 return ordering.Direction == OrderByDirection.Descending
-                    ? results.OrderBy(e => e.Name).OrderByDescending(e => e.Prices.Min(x => x.HourlyRate))
-                    : results.OrderBy(e => e.Name).OrderBy(e => e.Prices.Min(x => x.HourlyRate));
+                    ? results
+                        .OrderBy(e => e.Name)
+                        .OrderBy(s => (ordering.SeoUrlOrderBy == null || ordering.SeoUrlOrderBy.Length == 0) ? -1 : Array.IndexOf(ordering.SeoUrlOrderBy, s.SeoUrl))
+                        .OrderByDescending(e => e.Prices == null ? int.MinValue : e.Prices!.Min(x => x.HourlyRate))
+                    : results
+                        .OrderBy(e => e.Name)
+                        .OrderBy(s => (ordering.SeoUrlOrderBy == null || ordering.SeoUrlOrderBy.Length == 0) ? -1 : Array.IndexOf(ordering.SeoUrlOrderBy, s.SeoUrl))
+                        .OrderBy(e => e.Prices == null ? int.MaxValue : e.Prices!.Min(x => x.HourlyRate));
 
             case TuitionPartnerOrderBy.SeoList:
                 if (ordering.SeoUrlOrderBy == null || ordering.SeoUrlOrderBy.Length == 0)
                 {
                     return results;
                 }
-                return results.OrderBy(x => Array.IndexOf(ordering.SeoUrlOrderBy, x.SeoUrl));
+                return results
+                    .OrderBy(x => x.Prices == null ? (1000 + Array.IndexOf(ordering.SeoUrlOrderBy, x.SeoUrl)) : Array.IndexOf(ordering.SeoUrlOrderBy, x.SeoUrl));
```

Comment:
Can we please extract the `1000` from here and declare a variable with a meaningful name?

Reply:
Done